rasgotransforms/rasgotransforms/macros/filter.sql

```diff
@@ -16,19 +16,34 @@
 {% endfor %}
 (
     {% for filter in filters %}
+
     {% if filter is not string and filter is not mapping %}
     {% set filter = dict(filter) %}
     {% endif %}
+
     {% if 'columnName' in filter %}
         {% do filter.__setitem__('column_name', filter.columnName) %}
     {% endif %}
     {% if 'comparisonValue' in filter %}
         {% do filter.__setitem__('comparison_value', filter.comparisonValue) %}
     {% endif %}
+
+    {% if filter.comparison_value is mapping %}
+        {% if filter.comparison_value.type == 'RELATIVEDATE' or filter.comparison_value.type == 'relativedate' %}
+            {% if 'datePart' in filter.comparison_value %}
+                {% do filter.comparison_value.__setitem__('date_part', filter.datePart) %}
+            {% endif %}
+            {% if filter.comparison_value.direction == 'past' %}
+                {% do filter.comparison_value.__setitem__('offset', -filter.comparison_value.offset)  %}
+            {% endif %}
+            {% do filter.__setitem__('comparison_value', "(CURRENT_DATE + INTERVAL '" ~ filter.comparison_value.offset ~ filter.comparison_value.date_part ~ "')") %}
```

Comment:
I just used ANSII SQL here so we didn't have to translate between Snowflake/BQ/crap syntax. Open to other ways to handle this